### PR TITLE
Add progress bar for CTGAN fitting

### LINKED ISSUE #298: Add progress bar for CTGAN fitting (+ save the loss values)
### Description
Currently, when you toggle the CTGAN's `verbose=True` option, it starts printing out the loss values (for generator and discriminator) for each epoch.

```python
>>> ctgan = CTGAN(epochs=10, verbose=True)
>>> ctgan.fit()
Epoch 1, Loss G:  1.7863,Loss D: -0.3639
Epoch 2, Loss G:  1.5484,Loss D:  0.2260
Epoch 3, Loss G:  1.3633,Loss D: -0.0441
...
```

**Problems**
- The verbosity fills up my screen with values, so it's not the cleanest way to see the fitting progress (% completed). A progress bar would be nicer.
- I may want to reference the loss values later. It's cumbersome to try to capture these values as they are printed out.

### Expected behavior
**Progress bar**: When `verbose=True`, we should use `tqdm` to show a progress bar
1. The progress bar should increment at each epoch
2. The loss values should refresh at each epoch. They should be rounded to 2 decimal digits, in order to keep consistent formatting

Example
ctgan_pbar

**Save the loss values for later use**:  The CTGAN model object should save the full (unrounded) loss values so that the user can access them later. Save them as a pandas.DataFrame.

```
>>> ctgan.loss_values

   Epoch  Generator Loss  Distriminator Loss
0      1          1.7863             -0.3639
1      2          1.5484              0.2260
2      3          1.3633             -0.0441
```

### Additional context
Using TQDM, you can reset the description using the `set_description` method.

Example:
```python
print('Fitting CTGAN ...')

num_epochs = 10
pbar = tqdm(range(num_epochs))

description = 'Gen. ({gen:.2f}) | Discrim. ({dis:.2f})'
pbar.set_description(description.format(gen=0, dis=0))

for epoch in pbar:
    # after training epoch you'll have new loss values
    # generator_loss, discriminator_loss
    # save them, and the set the description 
    pbar.set_description(description.format(gen=generator_loss, dis=discriminator_loss))
```

### PR DESCRIPTION
**Pull request by frances-h**
Resolve #298

### CODE REVIEW

**Review by fealho — APPROVED** [review]

**Review by pvk-developer — APPROVED** [review]
LGTM!
